composer.json

```diff
@@ -22,6 +22,7 @@
     "require-dev": {
         "doctrine/coding-standard": "^11.1",
         "phpbench/phpbench": "^1.2",
+        "psr/log": "^1.1.4|^2|^3",
```

Comment:
We can add this to the `suggests` section so people know they can use a PSR logger.

Reply:
Can do. In case you missed @GromNaN's [previous comment](https://github.com/mongodb/mongo-php-library/pull/1158#discussion_r1324489025):

> We may mark the dependency as "compatible" in composer, one day. https://github.com/composer/composer/issues/11635

Reply:
I think this is a moot point. If it was just type hints for the add/removeSubscriber methods, we could stick with `suggestions`, but PsrLogAdapter references the PSR LogLevel constants directly in a private constant, so we actually depend on `psr/log` being available.

Reply:
This is still an optional dependency, because logging is an optional feature. You could throw an exception before the PsrLogAdapter class declaration, telling a psr-3 compatible logger needs to be installed.

Reply:
@GromNaN: I expect at some point PHPLIB will need to write its own log messages for specs, at which point we'd need to introduce the dependency then.

If you feel strongly about making it optional now and deferring the hard requirement until then, I can implement an exception if the PSR classes don't exist.

Reply:
Ok, let's stick with what's most standard with a required dependency.